guiEditor/src/components/propertyTab/propertyGrids/gui/ellipsePropertyGridComponent.tsx

```diff
@@ -47,6 +40,17 @@ export class EllipsePropertyGridComponent extends React.Component<IEllipseProper
                     target={makeTargetsProxy(ellipses, this.props.onPropertyChangedObservable)}
                     propertyName="thickness"
                     onPropertyChangedObservable={this.props.onPropertyChangedObservable}
+                    unit={"PX"}

```

Comment:
is `unit: 'PX'` and arbitrary value? or is there some sort of mapping between PX => functionality and % => functionality?

Reply:
For now, the unit string and the functionality of the unit button is totally controlled outside of the component itself. So in the common control component, we handle all the logic of swapping units between PX and %.

I could make some kind of more formal unit class or enum, but honestly, I didn't see much added value in it over just passing in a string. Especially since most controls will just use a fixed PX unit.

Reply:
That's a fair point. The only reason I see making an enum in that case is to make sure formatting (px vs PX vs Px) will be preserved. But that's fine like that.